### DISCUSSION TITLE
Issue with Invisible reCAPTCHA on Multi-Step Form Submissions

### DISCUSSION THREAD

**Discussion by Selvavinoth1** [solspace/craft-freeform#1755] [Questions & Answers]
### What happened?

Hi Team, 

We have identified an issue where submissions from the multi-step form are landing in the spam folder. Upon investigation, we noticed the following error on submissions flagged as spam: **The response parameter is invalid or malformed.**

This issue specifically occurs when using the Invisible reCAPTCHA functionality. Submissions made with reCAPTCHA v2 (visible) are correctly delivered to the legitimate folder, indicating the problem is isolated to the invisible reCAPTCHA implementation.

We have verified the reCAPTCHA keys and template-level configuration, and everything appears to be valid. However, we suspect the issue is related to reCAPTCHA validation during the submission process, particularly on the Freeform multi-step form.
![freefrom-captacha-validation](https://github.com/user-attachments/assets/c4f25b45-5d4e-4888-8db9-86d2a12a5d87)

### Errors and Stack Trace (if available)

```shell
reCAPTCHA - The response parameter is invalid or malformed.
```

### How can we reproduce this?

1. Submit a multistep form
2. Submission is landing into the spam folder

### Freeform Edition

Pro

### Freeform Version

5.7.2

### Craft Version

4.13.0

### When did this issue start?

Unsure

### Previous Freeform Version

5.7.1

**Comment by kjmartens** [comment]
Sorry about this @Selvavinoth1,

Can you confirm which reCAPTCHA version this is? E.g. reCAPTCHA v2 Invisible or v3?

Can you also take a screenshot of your Freeform Diagnostics page and post that here? 🙂

**Comment by Selvavinoth1** [comment]
Hi @kjmartens,

Thanks for your response, we're using reCAPTCHA V3 Invisible. Attached the screenshot of the Freeform Diagnostics page. 

![image](https://github.com/user-attachments/assets/7a4bd207-f08b-4be3-98cb-668b6ea24f7d)

Please check and let us know if any other details required from our end.

**Comment by kjmartens** [comment]
Thanks for the additional information.

I have played around with this on my end and I can't seem to get it to fail. Is it possible you have something configured incorrectly or differently inside the reCAPTCHA admin site? I'm aware there are additional settings in there, and I wonder if any of that has been tinkered with. Are you able to provide a screenshot of that (with keys censored of course)? 🙂

**Comment by Selvavinoth1** [comment]
Hi @kjmartens,

Thanks for your response, We have provided the screenshot of Freefrom reCAPTCHA configuration. Please check and confim with us. 

![Captchas - settings-page](https://github.com/user-attachments/assets/73b82871-4036-40bd-9320-d23f94954041)

**Comment by kjmartens** [comment]
Thanks for the extra screenshot, @Selvavinoth1.

Could you provide more information on the following:

- Can you provide a screenshot of the reCAPTCHA admin (from reCAPTCHA website, not Freeform settings).
   - This is what I am referring to:
      <img width="918" alt="Screenshot 2024-11-29 at 11 39 12 AM" src="https://github.com/user-attachments/assets/9bfc20a1-36ea-46c9-adf3-b6ac559530b8">
- A screenshot of the configuration of the reCAPTCHA integration inside of the form builder.
   - This is what I am referring to:
      <img width="1060" alt="Screenshot 2024-11-29 at 11 43 02 AM" src="https://github.com/user-attachments/assets/75d73691-2c92-464c-8abc-4032cfa34181">
- Are you seeing any JS or loading errors in the browser console? If so, can you provide those here?
   - One helpful thing is getting the Network Response of an error (if there is one). You can do it this way (keep in mind that you aren't looking for a row named `options`... it'll be different for yours and named something else, but be in red.
      <img width="853" alt="Network Response" src="https://github.com/user-attachments/assets/a786a851-4cd3-49f4-8e47-a668602f0ae1">

Thanks 🙂

**Comment by Selvavinoth1** [comment]
Hi @kjmartens,

Thanks for your response, we have provided the reCAPTCHA admin, form intergation and network error screenshot. Please note that we didn't encounter any JS errors on the browser console and network response. Attached the screenshots for your reference. Please check and confirm with us. 

![image](https://github.com/user-attachments/assets/d6c97ecf-f9b8-4e9b-baeb-25a852f62fa9)

![image](https://github.com/user-attachments/assets/0fa89b12-fa52-470f-aa71-8106d9326686)

![image](https://github.com/user-attachments/assets/1cffab2e-c9c0-4bd5-966d-69b8a7b585d7)

![image](https://github.com/user-attachments/assets/27c48236-ddd2-4df8-846d-9e222deaa331)

**Comment by Selva-infotex** [comment]
Hi @kjmartens, 

We wanted to follow up on the above request, please check and let us know if any other details are required from us to investigate further.

**Comment by kjmartens** [comment]
Hi @Selva-infotex / @Selvavinoth1,

Sorry for the delay...

I was able to finally reproduce a similar issue on my end, but it only happens if I am using reCAPTCHA v2 Checkbox and set it to send the submission to the spam folder (instead of present an error) - it's being bypassed by the user so it goes straight to spam but I guess reCAPTCHA doesn't have a valid reason for it.

You're getting this on v2 Invisible and v3, however, so this isn't exactly the answer.

I'm curious, are you able to share a live URL where this is happening? If you wish to not display it publicly here, can you send it in a private support ticket at support@solspace.com? 🙂

**Comment by Selvavinoth1** [comment]
Hi @kjmartens,

Thanks for the confirmation on the issue, yes, we're also faccing the issue with the  v2 and v3 Invisible reCAPTCHA. Furthermore, we'll dicuss with our internal team and provide you the website URL through the email shortly.

**Comment by Selvavinoth1** [comment]
Hi @kjmartens, 

We have shared the website URL through the [support@solspace.com](mailto:support@solspace.com) email address. Please check and let us know if any other details required from our end.
